pkg/transforms/policy.go

```diff
@@ -37,6 +41,106 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
 	if okns && okpp {
 		node.Properties["_parentPolicy"] = pnamespace + "/" + ppolicy
 	}
+
+	return &PolicyResource{node: node}
+}
+
+// For cert, config, operator policies.
+// This function returns `annotations`, `managedFields` for `source`,
+// and `severity`, `compliant`, `remediationAction`, `parentPolicyNs`, `parentPolicyName`
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	annos := c.GetAnnotations()
+	// This annotation is large and useless
+	delete(annos, "kubectl.kubernetes.io/last-applied-configuration")
+
+	node.Properties["annotations"] = annos
```

Comment:
- Annotations are common to all resources, so let's move this to `commonProperties()` in `common.go`
- I like that it excludes the last-applied-configuration, but I want to make it more generic to exclude any other large annotations. Maybe exclude any annotation with more than 64 characters?
- We'll need to update how we query for `annotation`. This will follow the same as `label`.

Reply:
@yiraeChristineKim @mprahl I want to resolve these before we merge. We have 2 options, 
- Ideally, I'd like to implement `annotation` to be common to any resource, but this adds the risk of increasing memory.
- Or we could `_annotation` to hide this property and avoid confusing users. My goal is that this will be only temporary while we implement a common `annotation`, so we'll have to update the client code later.

Reply:
@jlpadilla I actually need the annotations for the Gatekeeper constraints too, so if you're good with it, we can make this common. I'd rather not hide it since it could be useful.

Reply:
Great! My preference is to make `annotation` common, but it will require some additional work.
Everything else in this PR is good to merge. I suggest that we separate the common annotation to a new PR and merge this one.

Reply:
@yiraeChristineKim can you do something like this?

```diff
diff --git a/pkg/transforms/common.go b/pkg/transforms/common.go
index 57d97c0..64baaca 100644
--- a/pkg/transforms/common.go
+++ b/pkg/transforms/common.go
@@ -40,6 +40,16 @@ func commonProperties(resource v1.Object) map[string]interface{} {
        if resource.GetLabels() != nil {
                ret["label"] = resource.GetLabels()
        }
+
+       annotations := resource.GetAnnotations()
+
+       if annotations != nil {
+               // This annotation is large and useless
+               delete(annotations, "kubectl.kubernetes.io/last-applied-configuration")
+
+               ret["annotations"] = annotations
+       }
+
        if resource.GetNamespace() != "" {
                ret["namespace"] = resource.GetNamespace()
        }
@@ -196,7 +206,8 @@ func edgesByDestinationName(
        destKind string,
        nodeInfo NodeInfo,
        ns NodeStore,
-       seenDests []string) []Edge {
+       seenDests []string,
+) []Edge {
        ret := []Edge{}
        for _, value := range seenDests {
                // Checking against nodeInfo.UID - it gets updated every time edgesByDestinationName is called
@@ -268,7 +279,7 @@ func edgesByDestinationName(
                                        nodeInfo.EdgeType, destKind, nodeInfo.NameSpace+"/"+name)
                        }
                }
-               seenDests = append(seenDests, nodeInfo.UID) //add nodeInfo UID to processed/seen nodes
+               seenDests = append(seenDests, nodeInfo.UID) // add nodeInfo UID to processed/seen nodes
 
                // If the destination node has property _ownerUID, create an edge between the pod and the destination's owner
                // Call the edgesByOwner recursively to create the uses edge
@@ -378,7 +389,6 @@ func edgesByDeployerSubscriber(nodeInfo NodeInfo, ns NodeStore) []Edge {
        }
        ret = findSub(nodeInfo.UID)
        return ret
-
 }
 
 // Build edges from the source node in nodeInfo to all applications/channels in the subscription's metadata.
diff --git a/pkg/transforms/common_test.go b/pkg/transforms/common_test.go
index bf93898..48eed2d 100644
--- a/pkg/transforms/common_test.go
+++ b/pkg/transforms/common_test.go
@@ -18,8 +18,10 @@ import (
        machineryV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var labels = map[string]string{"app": "test", "fake": "true", "component": "testapp"}
-var timestamp = machineryV1.Now()
+var (
+       labels    = map[string]string{"app": "test", "fake": "true", "component": "testapp"}
+       timestamp = machineryV1.Now()
+)
 
 // Helper function for creating a k8s resource to pass in to tests.
 // In this case it's a pod.
@@ -32,11 +34,11 @@ func CreateGenericResource() machineryV1.Object {
        p.UID = "00aa0000-00aa-00a0-a000-00000a00a0a0"
        p.CreationTimestamp = timestamp
        p.Labels = labels
+       p.Annotations = map[string]string{"hello": "world"}
        return &p
 }
 
 func TestCommonProperties(t *testing.T) {
-
        res := CreateGenericResource()
        timeString := timestamp.UTC().Format(time.RFC3339)
 
@@ -46,6 +48,7 @@ func TestCommonProperties(t *testing.T) {
        AssertEqual("name", cp["name"], interface{}("testpod"), t)
        AssertEqual("namespace", cp["namespace"], interface{}("default"), t)
        AssertEqual("created", cp["created"], interface{}(timeString), t)
+       AssertEqual("annotations", cp["annotations"].(map[string]string)["hello"], "world", t)
 
        noLabels := true
        for key, value := range cp["label"].(map[string]string) {
diff --git a/pkg/transforms/genericResource_test.go b/pkg/transforms/genericResource_test.go
index 9adc9b4..496fb9d 100644
--- a/pkg/transforms/genericResource_test.go
+++ b/pkg/transforms/genericResource_test.go
@@ -5,7 +5,9 @@ package transforms
 import (
        "testing"
 
+       "github.com/stretchr/testify/assert"
        "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+       "k8s.io/apimachinery/pkg/util/sets"
 )
 
 func Test_genericResourceFromConfig(t *testing.T) {
@@ -19,11 +21,30 @@ func Test_genericResourceFromConfig(t *testing.T) {
        AssertEqual("namespace", node.Properties["namespace"], "open-cluster-management", t)
        AssertEqual("created", node.Properties["created"], "2023-08-23T15:54:22Z", t)
 
+       annotations, ok := node.Properties["annotations"].(map[string]string)
+       assert.True(t, ok)
+
+       // Ensure last-applied-configuration is not present
+       expectedAnnotationKeys := sets.New[string](
+               "alm-examples", "capabilities", "categories", "certified", "createdAt", "description", "olm.operatorGroup",
+               "olm.operatorNamespace", "olm.targetNamespaces", "operatorframework.io/initialization-resource",
+               "operatorframework.io/properties", "operatorframework.io/suggested-namespace",
+               "operators.openshift.io/infrastructure-features", "operators.openshift.io/valid-subscription",
+               "operators.operatorframework.io/internal-objects", "support",
+       )
+
+       actualAnnotationKeys := sets.Set[string]{}
+
+       for key := range annotations {
+               actualAnnotationKeys.Insert(key)
+       }
+
+       assert.True(t, expectedAnnotationKeys.Equal(actualAnnotationKeys))
+
        // Verify properties defined in the transform config
        AssertEqual("display", node.Properties["display"], "Advanced Cluster Management for Kubernetes", t)
        AssertEqual("phase", node.Properties["phase"], "Succeeded", t)
        AssertEqual("version", node.Properties["version"], "2.9.0", t)
-
 }
 
 func Test_genericResourceFromConfigVM(t *testing.T) {
@@ -40,5 +61,4 @@ func Test_genericResourceFromConfigVM(t *testing.T) {
        // Verify properties defined in the transform config
        AssertEqual("display", node.Properties["status"], "Running", t)
        AssertEqual("phase", node.Properties["ready"], "True", t)
-
 }
diff --git a/pkg/transforms/genericresource.go b/pkg/transforms/genericresource.go
index c752409..b3e17a4 100644
--- a/pkg/transforms/genericresource.go
+++ b/pkg/transforms/genericresource.go
@@ -99,6 +99,16 @@ func genericProperties(r *unstructured.Unstructured) map[string]interface{} {
        if r.GetLabels() != nil {
                ret["label"] = r.GetLabels()
        }
+
+       annotations := r.GetAnnotations()
+
+       if annotations != nil {
+               // This annotation is large and useless
+               delete(annotations, "kubectl.kubernetes.io/last-applied-configuration")
+
+               ret["annotations"] = annotations
+       }
+
        if r.GetNamespace() != "" {
                ret["namespace"] = r.GetNamespace()
        }
@@ -109,7 +119,6 @@ func genericProperties(r *unstructured.Unstructured) map[string]interface{} {
                ret["_hostingDeployable"] = r.GetAnnotations()["apps.open-cluster-management.io/hosting-deployable"]
        }
        return ret
-
 }
 
 func genericMetadata(r *unstructured.Unstructured) map[string]string {
diff --git a/pkg/transforms/policy.go b/pkg/transforms/policy.go
index c4ddc7b..5ad15f4 100644
--- a/pkg/transforms/policy.go
+++ b/pkg/transforms/policy.go
@@ -49,12 +49,6 @@ func PolicyResourceBuilder(p *policy.Policy) *PolicyResource {
 // This function returns `annotations`, `_isExternal` for `source`,
 // and `severity`, `compliant`, and `remediationAction`.
 func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
-       annos := c.GetAnnotations()
-       // This annotation is large and useless
-       delete(annos, "kubectl.kubernetes.io/last-applied-configuration")
-
-       node.Properties["annotations"] = annos
-
        node.Properties["_isExternal"] = false
 
        for _, m := range c.GetManagedFields() {
diff --git a/test-data/clusterserviceversion.json b/test-data/clusterserviceversion.json
index 4cc4b2e..c74927d 100644
--- a/test-data/clusterserviceversion.json
+++ b/test-data/clusterserviceversion.json
@@ -18,6 +18,7 @@
             "operators.openshift.io/infrastructure-features": "[\"disconnected\", \"proxy-aware\", \"fips\"]",
             "operators.openshift.io/valid-subscription": "[\"OpenShift Platform Plus\", \"Red Hat Advanced Cluster Management for Kubernetes\"]",
             "operators.operatorframework.io/internal-objects": "[]",
+            "kubectl.kubernetes.io/last-applied-configuration": "something",
             "support": "Red Hat"
         },
         "creationTimestamp": "2023-08-23T15:54:22Z",
```

Reply:
@jlpadilla here is the PR:
https://issues.redhat.com/browse/ACM-13031

Reply:
@jlpadilla here is the PR:
https://github.com/stolostron/search-collector/pull/302